R/src/feather-types.cpp

```diff
@@ -148,17 +148,25 @@ SEXP toSEXP(const PrimitiveArray* val) {
 
   case PrimitiveType::UTF8: {
     auto asChar = reinterpret_cast<const char*>(val->values);
+    auto total_bytes = meta.total_bytes;
     for (int i = 0; i < n; ++i) {
       uint32_t offset1 = val->offsets[i], offset2 = val->offsets[i + 1];
```

Comment:
This should be `int32_t` -- if the offsets are negative (due to data corruption) this will be a bad cast